server/routes/api/groups/groups.ts

```diff
@@ -35,6 +37,15 @@ router.post(
       limit: ctx.state.pagination.limit,
     });
 
+    if (userId) {
+      groups = groups.filter((group) => {
```

Comment:
This filter needs to be part of the query to function properly

Reply:
Well, I tried that initially using `include` and `where` combo but it's a bit tricky since `groupMemberships` are not retained then. For example, attempted with `scope` :point_down: 
```
include: [{
  association: "groupMemberships",
  required: false,
  where: {
    userId: query.userId
  }
}]
```
but it filters out `groupMemberships` other than that of `query.userId`

Next, attempted using [top-level clauses](https://sequelize.org/docs/v6/advanced-association-concepts/eager-loading/#complex-where-clauses-at-the-top-level) which resulted in same issue.

I assume we need to return all `groupMemberships` for the `group` irrespective of what `query.userId` is?

Reply:
I haven't tried it, but perhaps we can do the include twice with different names…

```
include: [{
  association: "groupMemberships",
  required: true, 
  where: { userId: query.userId }
  as: "tmp",
}]
```

Reply:
```
      {
        association: "groupMemberships",
        required: true,
        where: {
          userId: query.userId,
        },
        as: "tmp",
      },
```

throws :point_down: 

```
{
	"ok": false,
	"error": "internal_server_error",
	"status": 500,
	"message": "table name \"groupMemberships\" specified more than once"
}
```
and,

```
      {
        model: GroupUser,
        required: true,
        where: {
          userId: memberId,
        },
        as: "tmp",
      },
```

throws :point_down: 

```
{
	"ok": false,
	"error": "internal_server_error",
	"status": 500,
	"message": "group_user is associated to group using an alias. You've included an alias (tmp), but it does not match the alias(es) defined in your association (groupMemberships)."
}
```

What works is adding an extra `@HasMany` decorator with an alias(say, `members`) and then, modifying the second `include` as :point_down: 

```
      {
        association: "members",
        required: true,
        where: {
          userId: memberId,
        },
      },
```